tests/roles/pcp_cleanup/tasks/main.yaml

```diff
@@ -2,29 +2,49 @@
   ansible.builtin.shell: |
     {{ shell_header }}
     {{ oc_header }}
+    # Cleanup OpenStackControlPlane object
+    oc delete --ignore-not-found=true osctlplane -l core.openstack.org/openstackcontrolplane -n openstack
 
-    oc delete --ignore-not-found=true --wait=false openstackdataplanedeployment/openstack
-    oc delete --ignore-not-found=true --wait=false openstackdataplanedeployment/openstack-nova-compute-ffu
-    oc delete --ignore-not-found=true --wait=false openstackcontrolplane/openstack
-    oc patch openstackcontrolplane openstack --type=merge --patch '
-    metadata:
-      finalizers: []
-    ' || true
-
-    while oc get pod | grep rabbitmq-server-0; do
-        sleep 2
-    done
-    while oc get pod | grep openstack-galera-0; do
-        sleep 2
+    # Ensure that all pods in openstack namespace are deleted
+    while oc get pod -n openstack | grep -E 'rabbitmq-server-0|openstack-galera-0'; do
+      sleep 2;
     done
 
+    # Delete Control Plane CRD
+    oc delete crd --ignore-not-found=true openstackcontrolplanes.core.openstack.org
+
+    # Cleanup OpenStackDataplane objects
+    oc delete --ignore-not-found=true OpenStackDataPlaneDeployment --all -n openstack
+    oc delete --ignore-not-found=true OpenStackDataPlaneNodeSet --all -n openstack
+    oc delete --ignore-not-found=true OpenStackDataPlaneService --all -n openstack
+
+    # Delete Dataplane CRDs
+    oc delete crd --ignore-not-found=true openstackdataplanedeployments.dataplane.openstack.org
+    oc delete crd --ignore-not-found=true openstackdataplanenodesets.dataplane.openstack.org
+    oc delete crd --ignore-not-found=true openstackdataplaneservices.dataplane.openstack.org
+
+    # Delete Adoption pods
     oc delete --ignore-not-found=true --wait=false pod mariadb-copy-data
     oc delete --ignore-not-found=true --wait=false pvc mariadb-data
     oc delete --ignore-not-found=true --wait=false pod ovn-copy-data
-    oc delete --ignore-not-found=true secret osp-secret
-    oc delete issuer rootca-internal --ignore-not-found
-    oc delete secret rootca-internal --ignore-not-found
+
+    # Delete secrets
+    oc delete --ignore-not-found=true secret -l service-cert -n openstack
+    oc delete --ignore-not-found=true secret -l ca-cert -n openstack
+    oc delete --ignore-not-found=true secret -l osp-secret -n openstack
+    oc delete --ignore-not-found=true secret -l osdp-service -n openstack
+
+    oc delete --ignore-not-found=true secret -l keystone -n openstack
+    oc delete --ignore-not-found=true secret -l libvirt-secret -n openstack
+    oc delete --ignore-not-found=true secret -l octavia-ca-passphrase -n openstack
+
+    oc delete --ignore-not-found issuer rootca-internal
+    oc delete --ignore-not-found secret rootca-internal
+
     oc delete subscription cluster-observability-operator -n openshift-operators --ignore-not-found
+
+    # Delete openstack namespace
+    oc delete --ignore-not-found=true namespace openstack
```

Comment:
the overall cleanup process look good to me. Technically just doing the openstack namespace delete should be enough but I heard that is it is not always stable.